receiver/nsxtreceiver/client_mock_test.go

```diff
@@ -0,0 +1,163 @@
+// Copyright  The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package nsxtreceiver // import "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/nsxtreceiver"
+import (
+	context "context"
+	testing "testing"
+
+	mock "github.com/stretchr/testify/mock"
+
+	model "github.com/open-telemetry/opentelemetry-collector-contrib/receiver/nsxtreceiver/internal/model"
+)
+
+const (
+	transportNode1    = "0e7bd3f2-bd49-4fa2-b650-9e9bfcdad827"
+	transportNode2    = "f5045ed2-43ab-4a35-a2c5-d20c30a32292"
+	transportNodeNic1 = "vmk10"
+	transportNodeNic2 = "vmnic0"
+	managerNode1      = "b7a79908-9808-4c9e-bb49-b70008993fcb"
+	managerNodeNic1   = "eth0"
+	managerNodeNic2   = "lo"
+)
+
+// MockClient is an autogenerated mock type for the MockClient type
+type MockClient struct {
+	mock.Mock
+}
+
+// ClusterNodes provides a mock function with given fields: ctx
+func (_m *MockClient) ClusterNodes(ctx context.Context) ([]model.ClusterNode, error) {
```

Comment:
I believe this file was generated by Mockery which has this convention, but it's not in `mocks/` with the autogenerated header. Is that accurate @schmikei?

Reply:
Yep the base was created using mockery but I ended up moving it into the package so its not entirely reproducible from the mockery command I'm fine removing the prefix on it for the time being since its not auto-generate-able from mockery anyhow.